### DISCUSSION TITLE
Noise field in a solid target

### DISCUSSION THREAD

**Discussion by HaoPengYY** [SmileiPIC/Smilei#777] [Q&A]
Dear developers,

Hi. I am not reporting a bug, but a question about the 2D simulation. We are trying to simulate the situation that a relativistic electron beam flys through the center of  a round solid tube (in 2D, this might be a flat electron beam interacts with two flat target). But we find there are intense noise fields in the solid  target, even though we use a very high temporal and spatial resolution and a large number of particles per cell. In the figure below, you can find the noise field labeled in the green rectangle.

<img width="737" alt="image" src="https://github.com/user-attachments
[REBu235c.py.zip](https://github.com/user-attachments/files/17971470/REBu235c.py.zip)
/assets/4869e4ce-f8d1-40c3-ad41-b45aa2f71755">

And attached with the input script. Our question is that why there are such fields and how do we remove them? Thank you in advance.

**Comment by HaoPengYY** [comment]
[REBu235c.py.zip](https://github.com/user-attachments/files/17971591/REBu235c.py.zip)
![1](https://github.com/user-attachments/assets/0156587a-c420-4b99-a88f-374dd8ad68f1)

**Comment by mccoys** [comment]
Please do not use issues for questions

**Reply by HaoPengYY** [reply]
Sorry, I was not aware of that.

**Comment by mccoys** [comment]
Have you checked your resolution compared to the Debye length?

Note that the particles "feel" the fields averaged by a shape function. Use `Probe` if you want to get fields that particles see.

**Reply by HaoPengYY** [reply]
We have checked the resolution compared to the skin depth, but not to the Debye length. The plasma is initialized as cold plasma. I guessed the fields we saw are the numerical noises, which are quite large compared to the self field of the low-current electron beam.

You mean that the fields from the DiagFields and DiagProbe are different? This is interesting, because we did not see these noise fields in the 3D simulations, in which we use a DiaProbe to check the fields. While we usually do not put a 2D DiagProbe in a 2D simulation, we just use DiagFields.  But why is this difference?

**Reply by mccoys** [reply]
The Field diagnostic is just the values on the grid. The probes are actually fake particles that use an interpolation technique to obtain the field at their position. Check the documentation to find out about interpolation

**Reply by HaoPengYY** [reply]
Thanks, we followed your suggestion and it works, but there is something wrong with the magnetic field, we are checking.
